src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -665,6 +706,51 @@ GenericParameterValue genericParam
 				}
 				break;
 
+			//
+			// GetConstructor (Type[])
+			// GetConstructor (BindingFlags, Type[])
+			// GetConstructor (BindingFlags, Binder, Type[], ParameterModifier [])
+			// GetConstructor (BindingFlags, Binder, CallingConventions, Type[], ParameterModifier [])
+			//
+			case IntrinsicId.Type_GetConstructor: {
+					BindingFlags? bindingFlags;
+					if (calledMethod.HasParameterOfType(0, "System.Reflection.BindingFlags"))
+						bindingFlags = GetBindingFlagsFromValue (argumentValues[0]);
+					else
+						// Assume a default value for BindingFlags for methods that don't use BindingFlags as a parameter
+						bindingFlags = BindingFlags.Public | BindingFlags.Instance;
+
+					int? ctorParameterCount = calledMethod.GetParametersCount () switch {
+						1 => (argumentValues[0].AsSingleValue () as ArrayValue)?.Size.AsConstInt (),
+						2 => (argumentValues[1].AsSingleValue () as ArrayValue)?.Size.AsConstInt (),
+						4 => (argumentValues[2].AsSingleValue () as ArrayValue)?.Size.AsConstInt (),
+						5 => (argumentValues[3].AsSingleValue () as ArrayValue)?.Size.AsConstInt (),
+						_ => null,
+					};
+
+					// Go over all types we've seen
+					foreach (var value in instanceValue) {
+						if (value is SystemTypeValue systemTypeValue && !BindingFlagsAreUnsupported (bindingFlags)) {
```

Comment:
Does this also need handling for null/empty inputs?

Reply:
Not really - we could add it, but it's not necessary. We need special handling only for intrinsics which set return value, those which don't can have empty foreach loops, it doesn't cause a problem.

Not sure what's better - be consistent with unnecessary lines of code, or keep it shorter.

Reply:
Oh right, the difference is that it doesn't have a return value. My preference is consistency, but either way is fine.